rescript/get_unite.py

```diff
@@ -0,0 +1,154 @@
+    URL = query_data["data"][0]["attributes"]["media"][-1]["url"]
+    return URL
+
+
+def _unite_get_tgz(
+    url: str = None, download_path: str = None, retries: int = 10
+) -> str:
+    """Download compressed database"""
+    for retry in range(retries):
+        # Track downloaded size
+        file_size = 0
+        try:
+            response = requests.get(url, stream=True)
+            # Save .tgz file
+            unite_file_path = os.path.join(download_path, "unitefile.tar.gz")
+            with open(unite_file_path, "wb") as f:
+                for chunk in response.iter_content(chunk_size=8192):
+                    if chunk:
+                        f.write(chunk)
+                        file_size += len(chunk)
+            # Check if the downloaded size matches the expected size
+            if file_size == int(response.headers.get("content-length", 0)):
+                return unite_file_path  # done!
+            else:
+                raise ValueError("File download")
+        except ValueError:
+            print("File incomplete, on try " + str(retry))
+            if retry == retries - 1:
+                raise ValueError("File failed to download!")
+
+
+def _unite_get_artifacts(
+    tgz_file: str = None, cluster_id: str = None
+) -> (DataFrame, DNAIterator):
+    """
+    Find and import files with matching cluster_id from .tgz
+
+    Returns: Tuple containing tax_results and seq_results
+    """
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # Extract from the .tgz file
+        with tarfile.open(tgz_file, "r:gz") as tar:
+            # Keep only _dev files
+            members = [m for m in tar.getmembers() if "_dev" in m.name]
+            if not members:
+                raise ValueError("No '_dev' files found in Unite .tgz file")
+            for member in members:
+                # Keep only base name
+                member.name = os.path.basename(member.name)
+                tar.extract(member, path=tmpdirname)
+        # Find and import the raw files...
+        for root, dirs, files in os.walk(tmpdirname):
+            # ... with the matching cluster_id
+            filtered_files = [
+                f for f in files if f.split("_")[4] == cluster_id
+            ]
+            if not filtered_files or len(filtered_files) > 2:
+                raise ValueError(
+                    "Found "
+                    + str(len(filtered_files))
+                    + " files found with cluster_id = "
+                    + cluster_id
+                )
+            for file in filtered_files:
+                fp = os.path.join(root, file)
+                if file.endswith(".txt"):
+                    taxa = TaxonomyFormat(fp, mode="r").view(DataFrame)
+                elif file.endswith(".fasta"):
+                    seqs = DNAFASTAFormat(fp, mode="r").view(DNAIterator)
+    return taxa, seqs
+
+
+def get_unite_data(
+    version: str = "9.0",
+    taxon_group: str = "fungi",
```

Comment:
I'd suggest setting `taxon_group="eukaryotes"`, as we often have users finding many "unclassified fungi", when in fact they are not fungi, often invertebrates, algae, etc.... It is best that we default to using all eukaryotes, as the other taxa act as decoy / outgroup taxa. This will allow users to more easily remove non-fungal taxa.

Reply:
Ah, I see you have done this before

Done!